tests/cli/RunCest.php

```diff
@@ -633,4 +635,262 @@ public function runBootstrapInSuiteConfig(\CliGuy $I)
+            ->addMetaStep('Page\DemoPageObject: demo action1')
+            ->addStep("I don't see file found", 'thisFileDoesNotExist')
+            ->addStep("I don't see file found", 'thisFileAlsoDoesNotExist')
+            ->addMetaStep('Page\DemoPageObject: demo action1')
+            ->addStep("I don't see file found", 'thisFileDoesNotExist')
+            ->addStep("I don't see file found", 'thisFileAlsoDoesNotExist')
+            ->addMetaStep('Page\DemoPageObject: demo action2')
+            ->addStep("I don't see file found", 'thisFileAgainDoesNotExist')
+        ],
+        [
+          'testHtmlReportRegexBuilder' => (new TestHtmlReportRegexBuilder('CodeceptionIssue4413Cest', 'twoIdentialSubStepsWithAnotherSubStepInBetween'))
+            ->addMetaStep('Page\DemoPageObject: demo action1')
+            ->addStep("I don't see file found", 'thisFileDoesNotExist')
+            ->addStep("I don't see file found", 'thisFileAlsoDoesNotExist')
+            ->addMetaStep('Page\DemoPageObject: demo action2')
+            ->addStep("I don't see file found", 'thisFileAgainDoesNotExist')
+            ->addMetaStep('Page\DemoPageObject: demo action1')
+            ->addStep("I don't see file found", 'thisFileDoesNotExist')
+            ->addStep("I don't see file found", 'thisFileAlsoDoesNotExist')
+        ],
+        [
+          'testHtmlReportRegexBuilder' => (new TestHtmlReportRegexBuilder('CodeceptionIssue4413Cest', 'subStepFollowedByTwoIdentialSubSteps'))
+            ->addMetaStep('Page\DemoPageObject: demo action2')
+            ->addStep("I don't see file found", 'thisFileAgainDoesNotExist')
+            ->addMetaStep('Page\DemoPageObject: demo action1')
+            ->addStep("I don't see file found", 'thisFileDoesNotExist')
+            ->addStep("I don't see file found", 'thisFileAlsoDoesNotExist')
+            ->addMetaStep('Page\DemoPageObject: demo action1')
+            ->addStep("I don't see file found", 'thisFileDoesNotExist')
+            ->addStep("I don't see file found", 'thisFileAlsoDoesNotExist')
+        ]
+      ];
+    }
+
+  /**
+   * @group reports
+   *
+   * @dataProvider htmlReportRegexCheckProvider
+   *
+   * @param CliGuy               $I
+   * @param \Codeception\Example $example
+   * @param Scenario             $scenario
+   */
+  public function runHtmlCheckReport(\CliGuy $I, \Codeception\Example $example, Scenario $scenario)
+  {
+    if (version_compare(phpversion(), '7.0', '<')) {
+      $scenario->skip('This test fails due to another Codeception bug that only happens with PHP 5.6: the execution of single CEST test cases does not work');
+    }
+
+    /** @var TestHtmlReportRegexBuilder $testBuilder */
+    $testBuilder = $example['testHtmlReportRegexBuilder'];
+    $testClass = $testBuilder->getTestClass();
+    $testCase = $testBuilder->getTestCase();
+
+    $test = $testClass . ':' . $testCase;
+    $I->wantTo('execute tests with html output and check html (' . $test . ')');
```

Comment:
A small nitpick, but I think that it would be better to use `$I->wantTo('verify that all steps are rendered correctly')`

Reply:
I tried following the other existing test `RunCest.runHtml()` (https://github.com/Codeception/Codeception/blob/a6f9bea0abd200553d05ee8716bbb3783f100867/tests/cli/RunCest.php#L33) that was very similar: mine is doing the same in that it produces HTML report, but by contrast also checks/verify/asserts it...

Reply:
That test is testing that Codeception is able to generate html reports, your test is testing a very specific issue.

Reply:
OK: since it is specific to HTML reports this should be also part of this `wantTo` step argument, shouldn't it? So maybe something like `$I->wantTo('verify that all steps are rendered correctly in HTML report')`?